File: Java/PrimerEjercicio.java
import java.util.Random;
import java.util.Scanner;

public class PrimerEjercicio {

    public static void main(String[] args) {
        Random random = new Random();
        Scanner scanner = new Scanner(System.in);

        // Obtener la cantidad de estudiantes utilizando ValidadorEntero
        System.out.println("Ingresa la cantidad de estudiantes ");
        int cantAlumnos = validarNumeroEntero(scanner);

        int[] calificaciones = new int[cantAlumnos];
        for (int i = 0; i < calificaciones.length; i++) {
            calificaciones[i] = random.nextInt(101);
        }
        for (int i = 0; i < calificaciones.length; i++) {
            System.out.println("Las calificaciones son: " + calificaciones[i]);
        }

        int mayor = calificaciones[0];
        int menor = calificaciones[0];

        for (int nota : calificaciones) {
            if (nota < menor) {
                menor = nota;
            }
            if (nota > mayor) {
                mayor = nota;
            }
        }
        System.out.println("La nota más baja es: " + menor);
        System.out.println("La nota más alta es: " + mayor);
    }
    public static int validarNumeroEntero(Scanner scanner) {
        while (true) {
            try {
                return Integer.parseInt(scanner.next());
            } catch (NumberFormatException e) {
                System.out.println("Error: Debe ingresar una cantidad valida de alumnos.");
            }
        }
    }

}

File: Python/primer_ejercicio.py
from rich.console import Console
console = Console()
import random
from utils import es_numero_entero
def menor_mayor(calificaciones):
    mayor =max(calificaciones)
    menor =min(calificaciones)
    return mayor,menor

def notas():
    cant_alumnos = es_numero_entero(console.input("[bold green]Digite la cantidad de los estudiantes\n[bold green]"))
    calificaciones = [random.randint(0, 100) for _ in range(cant_alumnos)]
    console.print(f"[bold yellow]Las calificaciones son: {calificaciones}[bold yellow]")

    menor = menor_mayor(calificaciones)
    mayor, menor = menor_mayor(calificaciones)
    console.print(f"[italic red]La nota mas baja es: {menor}[italic red] :disappointed_face:")
    console.print(f"[italic blue]La nota mas alta es: {mayor}[italic blue] :smiley:")

if __name__ == "__main__":
    notas()